src/BloomBrowserUI/bookEdit/bloomField/bloomField.ts

```diff
@@ -0,0 +1,254 @@
+/// <reference path="../../lib/jquery.d.ts" />
+
+// We have a lot of trouble with the embedded Firefox's ContentEditable behavior. The chief problem is that FF *really* wants to just use <br>s,
+// which are worthless because you can't style them to, for example, do paragraph indents.
+// So this module is for various work-arounds. Eventually we should probably admit defeat and move to some existing open-source div editor.
+
+//Firefox adds a <BR> when you press return, which is lame because you can't put css styles on BR, such as indent.
+//Eventually we may use a wysiwyg add-on which does this conversion as you type, but for now, we change it when
+//you tab or click out.
+
+enum CursorPosition {start, end}
+
+class BloomField {
+
+    static ManageField(target:HTMLElement) {
```

Comment:
This is a very generic name. It could mean anything from "take control of this field by changing something before they start editing" to "clean up this field after they finish editing". A comment or better name indicating where in the field's lifetime it applies would help, I think.